### ISSUE TITLE
(MR/VR) MR/VR World sharing and locomotion thoughts  - TBD

### ISSUE THREAD

**Issue by Aktion** [fortellgames/bw_bugtracker#239]
Valuable thoughts from Discord to keep them tracked

Also i have somewhat of a concern regarding the MR/VR version of the game and world sharing.
I know i've spoke about how the worlds being shared would have things inside walls and how cross-platform sharing may not work very well for MR/VR

But if you're going to limit the sharing to just MR/VR version and it's worlds, then it could work even with different room layouts if the environment of the room is shown for other players similar to how it's in Holotanks when you remote play. Although i think when you play other people's worlds that take place in their room, there needs to be at least some type of teleport locomotion cos there will be instances where you won't be able to go to certain locations of the room either to your room being smaller than their's. I was thinking a lot about this and i think the MR/VR version of this game should be separate from PC, Mobile and Console.

Or better yet. why not when you select the envrionments, you can select "Room" environment which is the MR environment.
and the actual Blocksworld environments like Blocksterbay are strictly VR with Teleport locomotion to move around them

i know the hiding of walls exist but i find it only useful on the flat environment and not in things like blockster bay.
for things like Blockster bay i don't like how the world environment is inside your room. I've always played with the endless expanse in the MR/VR because of that.

but then the issue would be that certain worlds if they are using MR would be built for specific room with specific furniture in it.
Like there can be a scenario somebody makes where you have to platform up to a bed/table but these things wouldn't be for other players.  that's why i think the game should have some type of strict VR version with teleport locomotion for the normal environments and MR without the locomotion but when sharing the world, it shares the room layout as well

I hope that Blocksworld gets some type of strict VR mode where you can walk/teleport around the environments, because sometimes part of me wants that to be in the game.
i like the MR in the game but it needs work and it's often or not a thing that isn't motivating me to play the game cos i know that whatever i build will after another play session get scrambled into a furniture where i wouldn't be able to grab it from.
The locomotion doesn't really have to be that complicated. just some simple teleport locomotion where you point where you wanna go and be able to rotate the view with right thumbstick with rotation snap 45 degrees

**Comment by DaviveCZ** [comment]
Some thought i had about locomotion for the game i said on Discord.

one thing that you guys could do with stirct VR mode and locomotion. Is to take some inspiration from the VR version of a game called Discovery.
It's a Minecraft clone of some sort that has a "Tabletop" mode where if you hold the grip button, you move your camera around. and when the grip is held and you move the left stick around, it would move the entire world. with both grips you could rotate and scale the world.

**Comment by Aktion** [comment]
@DaviveCZ this [Discovery ](https://www.meta.com/experiences/discovery/3626919647423171/)you are mentionin ?
I was talked about about this [Discover](https://www.meta.com/experiences/discover/7041851792509764/) when we talked :)-

**Comment by DaviveCZ** [comment]
yup. the MC clone. That was the one i spoke off. i want the locomotion like it's there where you grab yourself around the air. similar to some 3D modeling VR apps

**Comment by Aktion** [comment]
> Some thought i had about locomotion for the game i said on Discord.
> 
> one thing that you guys could do with stirct VR mode and locomotion. Is to take some inspiration from the VR version of a game called Discovery. It's a Minecraft clone of some sort that has a "Tabletop" mode where if you hold the grip button, you move your camera around. and when the grip is held and you move the left stick around, it would move the entire world. with both grips you could rotate and scale the world.

I don't see grips in Discovery - I see teleport or jump. Is it some settings I have to enable?

**Comment by Aktion** [comment]
![image](https://github.com/user-attachments/assets/25cce5a2-e228-4e92-922d-046b865c3e30)

![image](https://github.com/user-attachments/assets/dac6e59c-97a4-4020-8253-56056872299b)

**Comment by DaviveCZ** [comment]
ah i see. i forgot about this. it's found in the beta version of the game. Switch the app to 0.1.9.7 (Beta) and go to tabletop mode (found in menu when you turn your palm up). I forgot that this is still in the beta branch of the game on Quest

**Comment by Aktion** [comment]
How to switch ? Its paid app now

**Comment by DaviveCZ** [comment]
![image](https://github.com/user-attachments/assets/0a1dc90c-5fad-431d-9532-bf7b6382a294)
either on phone app or the storepage like above

**Comment by Aktion** [comment]
Found settings - so beta 0.1 was earlier? They are changed way to move or its upcoming changes?

**Comment by DaviveCZ** [comment]
upcoming changes for update

**Comment by Aktion** [comment]
Okay, i didnt found a grip controls but tabletop mode just scrolling the heighmap with an analogue stick - please confirm we are talking about that? I didn't found scaling too

**Comment by DaviveCZ** [comment]
you have to hold grip and trigger to do the movement i'm talking about.
hold grip and trigger on both controllers for zoom.
that kind of movement would be good in the game for something like Blocksworld

**Comment by Aktion** [comment]
> you have to hold grip and trigger to do the movement i'm talking about. hold grip and trigger on both controllers for zoom. that kind of movement would be good in the game for something like Blocksworld

Yeah I can see it now. This is more for tabletop mode, not fps POV as we have in BW. Space Pirate Trainer I'm referring had this locomotion where you are standing still but world is moving while holding grips.

**Comment by DaviveCZ** [comment]
do you have a video of it in Space Pirate Trainer?
I'm unable to find it on Youtube. It looks like it has no locomotion unless it's a new thing added in a newer update.

**Comment by Aktion** [comment]
https://github.com/user-attachments/assets/08d3a4e1-700d-4022-859d-3d09aa76e472
